test/test_torch.py

```diff
@@ -12100,28 +12100,39 @@ def test_storage_multigpu(self, devices):
     @skipCPUIfNoLapack
     def test_lu(self, device):
         from common_utils import random_fullrank_matrix_distinct_singular_value as fullrank
+        from common_utils import random_matrix_distinct_singular_value as arbitraryrank
 
         def run_test(device, pivot):
             def run_subtest(matrix_size, batches, device, pivot):
-                a = fullrank(matrix_size, *batches).to(device)
+                if isinstance(matrix_size, int):
+                    rows = columns = matrix_size
+                    a = fullrank(matrix_size, *batches).to(device)
+                else:
+                    rows, columns = matrix_size
+                    a = arbitraryrank(rows, columns, *batches).to(device)
                 a_LU_info, pivots_info, info_ = a.lu(pivot=pivot, get_infos=True)
-                self.assertEqual(a_LU_info.size(), torch.Size(batches + (matrix_size, matrix_size)))
-                self.assertEqual(pivots_info.size(), torch.Size(batches + (matrix_size,)))
+                self.assertEqual(a_LU_info.size(), torch.Size(batches + (rows, columns)))
+                self.assertEqual(pivots_info.size(), torch.Size(batches + (min(rows, columns),)))
                 self.assertEqual(info_.size(), torch.Size(batches))
-                self.assertEqual(info_.abs().sum(), 0)
+                # If randomly generated input matrix is singular then
+                # info_ contains indices i such that U[i,i] == 0. This
+                # does not indicate a failure of computing LU
+                # factorization, so we require info.min() >= 0.
```

Comment:
I think this comment could be worded a bit differently, along the lines of:
```If a randomly generated input matrix is singular, then info_ contains indices i such that U[i, i] == 0. This however conveys that the factorization was successful albeit with a singular input. Therefore, we require info.min() >= 0```

I was slightly confused about what failure meant, but re-browsing through the changes made me understand this.

Reply:
Thanks for the suggestion! Applied.